internal/framework/base/schema.go

```diff
@@ -0,0 +1,102 @@
+package base
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/framework/validators"
+)
+
+// BaseModel describes common fields for all of the Terraform resource data models
+//
+// Ideally this struct would be imbedded into all Resources and DataSources.
+// However, the Terraform Plugin Framework doesn't support unmarshalling nested
+// structs. See https://github.com/hashicorp/terraform-plugin-framework/issues/242
```

Comment:
Struct embedding is now allowed in the TF Framework. Going to update the Password policy resource to use the common params

Reply:
Updated `password_policy` resource to use common embedded struct